test/functional/p2p_sendrecon.py

```diff
@@ -0,0 +1,168 @@
+        self.log.info('valid SENDRECON')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        peer.send_message(create_sendrecon_msg())
+        time.sleep(1)
+        assert peer.is_connected
+        self.log.info('second SENDRECON triggers a disconnect')
+        peer.send_message(create_sendrecon_msg())
+        time.sleep(1)
+        assert not peer.is_connected
+
+        self.log.info('SENDRECON with initiator=responder=0 triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_no_role = create_sendrecon_msg()
+        sendrecon_no_role.initiator = False
+        sendrecon_no_role.responder = False
+        peer.send_message(sendrecon_no_role)
+        time.sleep(1)
+        assert not peer.is_connected
+
+        self.log.info('SENDRECON with initiator=1 and responder=0 from inbound triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_wrong_role = create_sendrecon_msg()
+        sendrecon_wrong_role.initiator = False
+        sendrecon_wrong_role.responder = True
+        peer.send_message(sendrecon_wrong_role)
+        time.sleep(1)
+        assert not peer.is_connected
+
+        self.log.info('SENDRECON with initiator=0 and responder=1 from outbound triggers a disconnect')
+        peer = self.nodes[0].add_outbound_p2p_connection(SendReconSender(), p2p_idx=3, connection_type="outbound-full-relay")
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_wrong_role = create_sendrecon_msg()
+        sendrecon_wrong_role.initiator = True
+        sendrecon_wrong_role.responder = False
+        peer.send_message(sendrecon_wrong_role)
+        time.sleep(1)
+        assert not peer.is_connected
+
+        self.log.info('SENDRECON with version=0 triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_until(lambda: peer.version_received)
+        sendrecon_low_version = create_sendrecon_msg()
+        sendrecon_low_version.version = 0
+        peer.send_message(sendrecon_low_version)
+        time.sleep(1)
+        assert not peer.is_connected
+
+        self.log.info('SENDRECON after VERACK triggers a disconnect')
+        peer = self.nodes[0].add_p2p_connection(SendReconSender(manual_verack=True), send_version=True, wait_for_verack=False)
+        peer.wait_for_verack()
+        peer.send_message(msg_verack())
+        peer.send_message(create_sendrecon_msg())
+        time.sleep(1)
+        assert not peer.is_connected
+
```

Comment:
in d3cc218415f3a5f9a7170cda48a5e3938900832c:

Nice, I like the new tests. Are you missing a test case for when the peer sends more than 1 `sendrecon` ?